modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/upgrade/v1_1_0/ImageTypeContentUpgradeProcess.java

```diff
@@ -80,9 +80,13 @@ private void _copyJournalArticleImagesToJournalRepository()
 					long userId = PortalUtil.getValidUserId(
 						companyId, (Long)values[4]);
 
-					long folderId =
-						_journalArticleImageUpgradeHelper.getFolderId(
-							userId, groupId, resourcePrimKey);
+					long folderId = 0L;
+
+					synchronized (this) {
```

Comment:
Why is that synchronized block here? I may be wrong, but there cannot be more than one thread executing the same upgrade process.

Reply:
because of this: https://github.com/liferay-lima/liferay-portal/blob/8bc9d57192db0ef9831d45c263703f87689ae53a/modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/upgrade/v1_1_0/ImageTypeContentUpgradeProcess.java#L58

that is resulting in this error:
`2022-04-05 14:04:39.604 ERROR [ForkJoinPool-17-worker-11][JDBCExceptionReporter:234] Batch entry 0 insert into DLFolder (mvccVersion, ctCollectionId, uuid_, groupId, companyId, userId, userName, createDate, modifiedDate, repositoryId, mountPoint, parentFolderId, treePath, name, description, lastPostDate, defaultFileEntryTypeId, hidden_, restrictionType, lastPublishDate, status, statusByUserId, statusByUserName, statusDate, folderId) values (0, 0, '12b088e7-d3d6-3de4-986e-90156bfe2f7c', 20142, 20115, 20119, NULL, '2022-04-05 14:04:39.525+00', '2022-04-05 14:04:39.525+00', 43309, 'TRUE', 0, '/43311/', 'com.liferay.journal', NULL, '2022-04-05 14:04:39.492+00', 0, 'TRUE', 0, NULL, 0, 0, NULL, NULL, 43311) was aborted: ERROR: duplicate key value violates unique constraint "ix_c7e346d2"_  Detail: Key (groupid, parentfolderid, name, ctcollectionid)=(20142, 0, com.liferay.journal, 0) already exists.  Call getNextException to see other errors in the batch. [Sanitized]
2022-04-05 14:04:39.604 ERROR [ForkJoinPool-17-worker-11][JDBCExceptionReporter:234] ERROR: duplicate key value violates unique constraint "ix_c7e346d2"_  Detail: Key (groupid, parentfolderid, name, ctcollectionid)=(20142, 0, com.liferay.journal, 0) already exists. [Sanitized]`


We can create a different LPS but I considered it not necessary to do that because both errors come together in the same Upgrade process...

Reply:
BTW I tested this solution and it fixes the problem with the index